### PR TITLE
[DOCS] Document `_mvt` API

### LINKED ISSUE #75242: [DOCS] Document vector tiles API
Adds formal REST API documentation for the `_mvt` endpoint.

Relates to #58696

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the `_mvt` API endpoint added with #73872 and #75522.

Closes #75242.

### Preview

https://elasticsearch_75384.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-vector-tile-api.html

### CODE REVIEW

File: docs/reference/search/vector-tile-api.asciidoc
Comment (jrodewig) [inline docs/reference/search/vector-tile-api.asciidoc:2]: I went with "Vector tile" as the API name. Let me know if you prefer something different.

File: rest-api-spec/src/main/resources/rest-api-spec/api/vector_tile.json
Comment (jrodewig) [inline rest-api-spec/src/main/resources/rest-api-spec/api/vector_tile.json:2]: I can change the name of this file if wanted.

File: docs/reference/search/vector-tile-api.asciidoc
Comment (jrodewig) [inline docs/reference/search/vector-tile-api.asciidoc:324]: This could be wrong.  I intuited this but couldn't find any reference to this field in https://github.com/mapbox/vector-tile-spec. Let me know if I'm off base here.

File: docs/reference/search.asciidoc
Comment (jrodewig) [inline docs/reference/search.asciidoc:47]: These `ifdef`/`endif` tags ensure these docs are only available on 7.x and master. These tags will hide the docs on 7.15 and other branches until removed. This is common practice for docs under a feature flag.

File: rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json
Comment (delvedor) [inline rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: Can you add the returned content type as well?

Comment (sethmlarson) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: Mimetype appears to be `application/vnd.mapbox-vector-tile`:

http://www.iana.org/assignments/media-types/application/vnd.mapbox-vector-tile

Comment (jrodewig) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: Thanks for the suggestion. Added with 105e36c06398d0ea95d46760e4b87bc4c12fce3f.

@iverase Do you mind double-checking me here?

Per https://github.com/mapbox/vector-tile-spec/issues/48 and http://www.iana.org/assignments/media-types/media-types.xhtml, the MIME type should be `application/vnd.mapbox-vector-tile`.

Comment (sethmlarson) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: My timing is on point! :P

Comment (jrodewig) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: Looks right:
https://github.com/elastic/elasticsearch/blob/bff790011c039f0c8777c55960c6b8141c0aaf3b/x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/RestVectorTileAction.java#L78

Comment (iverase) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search_mvt.json:9]: Yes, that is the mime type we return.

File: docs/reference/search/vector-tile-api.asciidoc
Comment (iverase) [reply docs/reference/search/vector-tile-api.asciidoc]: That's correct :)

**Comment by iverase** [comment]
I have a quick look and it is looking great. I would like to wait until next week to give further feedback as we are going to discuss about some rough edges. Hope that's ok.

**Comment by jrodewig** [comment]
Sounds good. Thanks @iverase!

File: docs/reference/search/search-vector-tile-api.asciidoc
Comment (iverase) [inline docs/reference/search/search-vector-tile-api.asciidoc:66]: Maybe we should explicitly say that fields should be indexed with doc values and cannot be nested fields?

Comment (jrodewig) [reply docs/reference/search/search-vector-tile-api.asciidoc]: Thanks for the suggestion. I added this with 68dabc1fbdc3acc6ba2de6573de97b8fed175914.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/search/search-vector-tile-api.asciidoc
Comment (jrodewig) [reply docs/reference/search/search-vector-tile-api.asciidoc]: Thanks @iverase. That does make sense. I pushed https://github.com/elastic/elasticsearch/pull/75384/commits/8dc8b3f9fa4c14a127bab96852e8b6b77c6fe9b1 to better describe the default sort.

File: docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc
Comment (jrodewig) [inline docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc:53]: This removes some duplicated copy for the `fields` API/parameter and replaces it with an xref. This should make it easier to maintain the docs if we add/remove more `format` options in the future.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by iverase — APPROVED** [review]
LGTM! Thank you so much @jrodewig, documentation is looking great and I think we can. already push this version forward. Note that we are in the process to remove the feature flag so might want to do the same before pushing it.

**Comment by jrodewig** [comment]
Thanks @iverase!

> we are in the process to remove the feature flag so might want to do the same before pushing it.

Thanks for letting me know. I'll wait until #76068 is merged before merging this PR.

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**
